docs(GcpNfsBackupSchedule): user documentation (#614)
kyma-project/cloud-manager: docs/user/resources/04-30-30-gcp-nfs-backup-schedule.md

# GcpNfsBackupSchedule Custom Resource

> The `gcpnfsbackupschedule.cloud-resources.kyma-project.io` custom resource (CR) represents the user-defined schedule for creating backup

Reviewer: ```suggestion
The `gcpnfsbackupschedule.cloud-resources.kyma-project.io` custom resource (CR) represents the user-defined schedule for creating a backup
```

> of the `GcpNfsVolume` instances at regular intervals. 

Reviewer 2: ```suggestion
of the `GcpNfsVolume` instances at regular intervals. It performs the following actions:
```

> of the `GcpNfsVolume` instances at regular intervals. 

Reviewer: ```suggestion
of the `GcpNfsVolume` instances at regular intervals.  The CR:
```

> - Creates the backups by creating the `gcpnfsvolumebackup.cloud-resources.kyma-project.io` resources on the specified interval.

Reviewer 2: ```suggestion
- Creates the backups by creating the `gcpnfsvolumebackup.cloud-resources.kyma-project.io` resources at the specified interval.
```

> - Allows users to specify days and times in the form of CRON expressions to automatically create the backups. 

Reviewer 2: ```suggestion
- Enables you to specify days and times in the form of CRON expressions to automatically create the backups. 
```

> - Automatically deletes the backups when the backup reaches the configured `maximum retention days`. 

Reviewer 2: ```suggestion
- Automatically deletes the backups when the backup reaches the configured maximum retention days value. 
```

Reviewer 2: If "maximum retention days" is a field/parameter, use bold font for it. If not, no special formatting is necessary. For more formatting guidelines, please see [03-formatting.md](https://github.com/kyma-project/community/blob/main/docs/guidelines/content-guidelines/03-formatting.md).

> - Allows the users to temporarily `suspend` or `resume` the backup creation/deletion. 

Reviewer 2: ```suggestion
- Enables you to temporarily suspend or resume the backup creation/deletion. 
```

Reviewer 2: If "suspend" and "resume" are UI elements, use bold font for them. Otherwise, no special formatting is necessary.

|-----------------------------|---------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| **nfsVolumeRef**            | object              | Required. GcpNfsVolume reference.                                                                                                                                                                                                                                                   |
| **nfsVolumeRef.name**       | string              | Required. Name of the existing GcpNfsVolume.                                                                                                                                                                                                                                        |

> | **nfsVolumeRef.namespace**  | string              | Optional. Namespace of the existing GcpNfsVolume.  Defaults to the namespace of the GcpNfsBackupSchedule resource if not provided.                                                                                                                                                  |

Reviewer: ```suggestion
| **nfsVolumeRef.namespace**  | string              | Optional. The namespace of the existing GcpNfsVolume.  Defaults to the namespace of the GcpNfsBackupSchedule resource if not provided.                                                                                                                                                  |
```

> | **location**                | string              | Optional. The Region where backup resides. Defaults to the region of source GcpNfsVolume.                                                                                                                                                                                           |

Reviewer: ```suggestion
| **location**                | string              | Optional. The region where backup resides. Defaults to the region of the source GcpNfsVolume.                                                                                                                                                                                           |
```

> | **schedule**                | string              | Optional. CRON type expression for the schedule. When this value is empty or not specified, this schedule runs only once at the specified start time. See also [Schedule Syntax](https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/#schedule-syntax)              |

Reviewer 2: ```suggestion
| **schedule**                | string              | Optional. CRON type expression for the schedule. When this value is empty or not specified, this schedule runs only once at the specified start time. See also [Schedule Syntax](https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/#schedule-syntax).              |
```

| **prefix**                  | string              | Optional. Prefix for the name of the created `GcpNfsVolumeBackup` resources. Defaults to name of this schedule.                                                                                                                                                                     |
| **startTime**               | metav1.Time         | Optional. Start time for the schedule. Value cannot be from the past. When not specified, the schedule becomes effective immediately.                                                                                                                                               |
| **endTime**                 | metav1.Time         | Optional. End time for the schedule. Value cannot be from the past or before the `startTime`. When not specified, the schedule runs indefinitely.                                                                                                                                   |

> | **maxRetentionDays**        | int                 | Optional. Maximum number of days to retain the backup resources. If not specified, the backups are retained indefinitely. If `deleteCascade` is true for this schedule, then all the backups will be deleted when the schedule is deleted irrespective of this configuration value. |

Reviewer: ```suggestion
| **maxRetentionDays**        | int                 | Optional. Maximum number of days to retain the backup resources. If not specified, the backups are retained indefinitely. If `deleteCascade` is `true` for this schedule, then all the backups will be deleted when the schedule is deleted irrespective of this configuration value. |
```

> | **suspend**                 | boolean             | Optional. Specifies whether or not to suspend the schedule temporarily. Defaults to false.                                                                                                                                                                                          |

Reviewer: ```suggestion
| **suspend**                 | boolean             | Optional. Specifies whether or not to suspend the schedule temporarily. Defaults to `false`.                                                                                                                                                                                          |
```

> | **deleteCascade**           | boolean             | Optional. Specifies whether to cascade delete the backup resources when this schedule is deleted. Defaults to false.                                                                                                                                                                |

Reviewer: ```suggestion
| **deleteCascade**           | boolean             | Optional. Specifies whether to cascade delete the backup resources when this schedule is deleted. Defaults to `false`.                                                                                                                                                                |
```
